File: backend/models/Video.js
// Video.js
const mongoose = require('mongoose');

const videoSchema = new mongoose.Schema({
  titulo: { type: String, required: true, trim: true, maxlength: 100 },
  url: { type: String, required: true, unique: true },
  thumbnail: { type: String } // Ya no necesitamos el 'default' aquí
});

// Usamos un hook 'pre-save' para generar la miniatura automáticamente
videoSchema.pre('save', function (next) {
  const videoId = this.url.match(/(?:v=|\/)([a-zA-Z0-9_-]{11})/)?.[1];
  if (videoId) {
    this.thumbnail = `https://img.youtube.com/vi/${videoId}/hqdefault.jpg`;
  }
  next();
});

module.exports = mongoose.model('Video', videoSchema);

File: frontend/lib/models/video.dart
class Video {
  final String id;
  final String titulo;
  final String url;
  final String thumbnail;

  Video({
    required this.id,
    required this.titulo,
    required this.url,
    required this.thumbnail,
  });

  factory Video.fromJson(Map<String, dynamic> json) {
    return Video(
      id: json['_id'] ?? '',
      titulo: json['titulo'] ?? '',
      url: json['url'] ?? '',
      thumbnail: json['thumbnail'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {

      'titulo': titulo,
      'url': url,
      'thumbnail': thumbnail,
    };
  }
}
